### PR TITLE
net: lib: nrf_cloud_coap: enable non-modem build

### PR DESCRIPTION
**Pull request by jayteemo**
Enable experimental Wi-Fi build option for nRF Cloud CoAP.

### CODE REVIEW

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c
Comment (glarsennordic) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c:454]: Is all of this original to this PR, or brought in from somewhere else in the SDK?

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c
Comment (glarsennordic) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c:43]: I agree, I don't think this is necessary

But I do want to understand what happens if `nrf_cloud_jwt_generate` is called without it. Do we just get an otherwise normal JWT that'll be rejected due to the invalid date?

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c
Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c:454]: it is new

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c
Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c:43]: this has not been removed, only moved.

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c
Comment (glarsennordic) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c:454]: Got it. I'll give it a more detailed review, then, unless you'd prefer I hold off for the next revision

Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c:454]: please review.
i have no major revisions planned, except for changing the overlay to align with https://github.com/nrfconnect/sdk-nrf/pull/13841.

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c
Comment (glarsennordic) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c:118]: The one thing I wonder is whether we should perhaps use mbedtls and psa crypto functions instead of tinycrypt, since we are targeting mbedtls on non-modem builds anyways

In the short term, this would save you having to manually parse PEM/DER. In the long term, this would save you having to refactor this code (as much) when we switch to opaque keys

It's not 100% clear to me how opaque keys will be handled in this case. But I know it will need to involve using mbedtls/psa crypto, so switching to that now will knock one thing off the list.

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c
Comment (glarsennordic) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c:195]: It might be more accurate to say `from the TLS credentials subsystem` here

Also, this will need a major refactor once we switch to using opaque, non-exportable private keys.

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c
Comment (glarsennordic) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c:207]: Technically, the keys stored here may be either in PEM format or in DER format. MbedTLS figures it out automatically based on whether the key is NULL-terminated. You can either do this too, or let MbedTLS handle it for you. Third option is to ignore it for now, but we'll be locking ourselves out of using DER credentials, and this will potentially create a confusing error for anyone who tries to.

**Review by glarsennordic — COMMENTED** [review]
I have reviewed the crypto stuff now -- overall seems seems fine, but I do recommend switching to mbedTLS. `mbedtls_ecdsa_write_signature` would be the right place to look I think? Assuming I am correct that you are using ecdsa here

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c
Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c:118]: this PR is already a refactor of my [original effort](https://github.com/nrfconnect/sdk-nrf/compare/main...jayteemo:fw-nrfconnect-nrf:IRIS-5723_nrf_jwt).  
this is just to get something working until PSA is available.
i view this as a temporary/experimental solution.
i'll let @plskeggs decide how much more work needs to be put into this.

Comment (plskeggs) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_jwt.c:118]: Let's keep it as is and revisit when switching to opaque.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:441]:
```suggestion
    * An overlay file :file:`overlay_nrf7002ek_wifi_coap_no_lte.conf` for experimental CoAP over Wi-Fi support.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:952]:
```suggestion
    * Support for specific credentials provisioning using the following Kconfig options:

      * :kconfig:option:`CONFIG_NRF_CLOUD_PROVISION_CA_CERT`
      * :kconfig:option:`CONFIG_NRF_CLOUD_PROVISION_CLIENT_CERT`
      * :kconfig:option:`CONFIG_NRF_CLOUD_PROVISION_PRV_KEY`

```

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/cellular/nrf_cloud_multi_service/overlay_nrf7002ek_wifi_coap_no_lte.conf
Comment (plskeggs) [inline samples/cellular/nrf_cloud_multi_service/overlay_nrf7002ek_wifi_coap_no_lte.conf:88]: do the commented out ones need to stay?

**Review by peknis — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]
Looks good.

**Review by glarsennordic — APPROVED** [review]

**Pull request merged**
